assembly/src/procedures/mod.rs

```diff
@@ -244,6 +246,14 @@ impl From<[u8; ProcedureId::SIZE]> for ProcedureId {
 
 impl From<&LibraryPath> for ProcedureId {
     fn from(path: &LibraryPath) -> Self {
+        let penultimate = path.num_components().saturating_sub(2);
+        let path =
+            if path.num_components() >= 2 && path.at_pos(penultimate) == Some(LibraryPath::MOD) {
+                path.remove_at(penultimate)
+                    .expect("could not remove penultimate path component")
+            } else {
+                path.clone()
+            };
         ProcedureId::new(path)
```

Comment:
I'd prefer to avoid path manipulations here. Ideally, the changes should not affect anything outside of `/src/library` modules. Basically, by the time we are here, module paths should have been adjusted so that `x/y/mod.masm` becomes `x::y`